tests/test_server.py

```diff
@@ -144,29 +145,68 @@ def test_ready_endpoint(client, redis_mock):
     assert result == "ok", f"Expected status = ok, but instead returned {result}."
 
 
-@pytest.mark.use_openai
-@pytest.mark.use_server
-def test_streaming(client):
-
-    data = pd.DataFrame.from_records(
+@pytest.mark.parametrize("input_data, skills, output_column", [
+    # text classification
+    (
         [
             {"task_id": 1, "text": "anytexthere", "output": "Feature Lack"},
             {"task_id": 2, "text": "othertexthere", "output": "Feature Lack"},
             {"task_id": 3, "text": "anytexthere", "output": "Feature Lack"},
             {"task_id": 4, "text": "othertexthere", "output": "Feature Lack"},
-        ]
+        ],
+        [{
+            "type": "ClassificationSkill",
+            "name": "text_classifier",
+            "instructions": "Always return the answer 'Feature Lack'.",
+            "input_template": "{text}",
+            "output_template": "{output}",
+            "labels": {
+                "output": [
+                    "Feature Lack",
+                    "Price",
+                    "Integration Issues",
+                    "Usability Concerns",
+                    "Competitor Advantage",
+                ]
+            },
+        }],
+        "output"
+    ),
+    # entity extraction
+    (
+        [
+            {"task_id": 1, "text": "John Doe, 26 years old, works at Google", "entities": [{"start": 0, "end": 8, "label": "PERSON"}, {"start": 26, "end": 36, "label": "AGE"}, {"start": 47, "end": 53, "label": "ORG"}]},
+            {"task_id": 2, "text": "Jane Doe, 30 years old, works at Microsoft", "entities": [{"start": 0, "end": 8, "label": "PERSON"}, {"start": 26, "end": 36, "label": "AGE"}, {"start": 47, "end": 55, "label": "ORG"}]},
+            {"task_id": 3, "text": "John Smith, 40 years old, works at Amazon", "entities": [{"start": 0, "end": 10, "label": "PERSON"}, {"start": 28, "end": 38, "label": "AGE"}, {"start": 49, "end": 55, "label": "ORG"}]},
+            {"task_id": 4, "text": "Jane Smith, 35 years old, works at Facebook", "entities": [{"start": 0, "end": 10, "label": "PERSON"}, {"start": 28, "end": 38, "label": "AGE"}, {"start": 49, "end": 57, "label": "ORG"}]},
```

Comment:
Mind just making the NER stuff a separate test? I know there'll be quite a bit of overlap, but would be nice to have different tests fail for different issues (as much as possible)

Reply:
but parameterize mark would make them as separate tests, that you can launch with -k individually

Reply:
yeah, it's not about running them individually so much as being able to assert on different properties of their outputs, as each skill can have its own special failure conditions like the unique individuals they are :butterfly: 

that being said, there is more refactoring in DIA-1324 to make these skills behave more similarly anyway, so we should unblock and resume discussion there. @pakelley